app/Http/Traits/UploadFileTrait.php

```
@@ -2,30 +2,41 @@
 {
     public function uploadFile($file, $fileLocation, $newName = null)
     {
-        $hashmake = Hash::make('secret');
-        $hashname = substr(hash('sha256', $hashmake), 0, 20);
-        $originalName = $newName ?? $file->getClientOriginalName();
-        $file->storePubliclyAs($fileLocation . $hashname . "/", $originalName, 'public');
-        return $fileLocation . $hashname . "/" . $originalName;
+        if (!file_exists($fileLocation)) {
+            Storage::disk('public')->makeDirectory($fileLocation);
+        }
+        $hashname = Str::random(20);
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Str' in method 'uploadFile'. (undefined)

(StaticAccess)